packages/grafana-prometheus/src/components/monaco-query-field/monaco-completion-provider/completions.ts

```diff
@@ -74,9 +87,16 @@ function getAllMetricNamesCompletions(dataProvider: DataProvider): Completion[]
   return dataProvider.metricNamesToMetrics(metricNames).map((metric) => ({
     type: 'METRIC_NAME',
     label: metric.name,
-    insertText: metric.name,
     detail: `${metric.name} : ${metric.type}`,
     documentation: metric.help,
+    ...(metric.isUtf8
+      ? {
+          insertText: `{"${metric.name}"${snippetMarker}}`,
+          insertTextRules: InsertAsSnippet,
```

Comment:
With respect to my earlier comment:
> Autocomplete was incorrect when inside of quotes (see ""label.with.spaß"=" in the first video, which was correct after starting over without being inside of quotes)

, it seems like that's an existing issue with our handling of label name autocomplete, unrelated to the UTF-8 work (but perhaps more easily exposed by the UTF-8 work). Handling the `situation.betweenQuotes` with label values could be a nice improvement in the future.